pennylane/ops/qubit/matrix_ops.py

```diff
@@ -313,11 +352,39 @@ def compute_decomposition(D, wires):
 
         **Example:**
 
+        # TODO: Update once Identity has been replaced by GlobalPhase
         >>> qml.DiagonalQubitUnitary.compute_decomposition([1, 1], wires=0)
-        [QubitUnitary(array([[1, 0], [0, 1]]), wires=[0])]
+        [QubitUnitary(array([[0.36235775+0.93203909j, 0.        +0.j        ],
+                             [0.        +0.j        , 0.36235775+0.93203909j]]), wires=[0]),
+         RZ(1.5000000000000002, wires=[1]),
+         RZ(-0.10000000000000003, wires=[0]),
+         IsingZZ(0.2, wires=[0, 1])]
 
         """
-        return [QubitUnitary(DiagonalQubitUnitary.compute_matrix(D), wires=wires)]
+        n = len(wires)
+        phases = qml.math.real(qml.math.log(D) * (-1j))
+        coeffs = _walsh_hadamard_transform(phases, n).T
+        global_phase = qml.math.exp(1j * coeffs[0])
+        # For all other gates, there is a prefactor -1/2 to be compensated.
+        coeffs = coeffs * (-2.0)
+
+        ops = [QubitUnitary(qml.math.tensordot(global_phase, qml.math.eye(2), axes=0), wires[0])]
+        for wire0 in range(n):
+            # Single PauliZ generators correspond to the coeffs at powers of two
+            ops.append(qml.RZ(coeffs[1 << wire0], n - 1 - wire0))
+            # Double PauliZ generators correspond to the coeffs at the sum of two powers of two
+            ops.extend(
+                qml.IsingZZ(coeffs[(1 << wire0) + (1 << wire1)], [n - 1 - wire0, n - 1 - wire1])
+                for wire1 in range(wire0)
+            )
```

Comment:
I like this part specifying `RZ` and `IsingZZ`. They are faster that `MultiRZ` in general.